src/iarc7_motion/state_monitor.py

```diff
@@ -0,0 +1,164 @@
+#!/usr/bin/env python
+import actionlib
+import rospy
+import sys
+import threading
+import traceback
+
+from geometry_msgs.msg import TwistStamped
+from geometry_msgs.msg import Twist
+from nav_msgs.msg import Odometry
+
+from iarc7_msgs.msg import TwistStampedArray, OdometryArray
+from iarc7_msgs.msg import FlightControllerStatus
+
+from iarc_tasks.takeoff_task import TakeoffTask
+from iarc_tasks.land_task import LandTask
+from iarc_tasks.test_task import TestTask
+from iarc_tasks.xyztranslation_task import XYZTranslationTask
+from iarc_tasks.track_roomba_task import TrackRoombaTask
+from iarc_tasks.hit_roomba_task import HitRoombaTask
+from iarc_tasks.block_roomba_task import BlockRoombaTask
+from iarc_tasks.hold_position_task import HoldPositionTask
+from iarc_tasks.height_recovery_task import HeightRecoveryTask
+from iarc_tasks.velocity_task import VelocityTask
+
+
+class StateMonitor:
+
+    def __init__(self):
+        # state data
+        self._roombas = None
+        self._obstacles = None
+        self._drone_odometry = None
+        self._arm_status = None
+        
+        # current state of HLM
+        self._initialized = False
+        self._waiting_on_takeoff = True
```

Comment:
I don't think this is how we want to track the last state. We probably want an enum?

Reply:
Didn't think of that, but that makes sense. Will be fixed.